dbt_coding_conventions.md

```diff
@@ -22,18 +16,29 @@
 }}
 ```
 
-## Base Models
 
-- Only base models should select from source tables / views
-- Only a single base model should be able to select from a given source table / view.
-- Base models should be placed in a base/ directory
-- Base models should perform all necessary data type casting
-- Base models should perform all field naming to force field names to conform to standard field naming conventions
-- Source fields that use reserved words must be renamed in base models
+## dbt conventions
+* Only `stg_` models (or `base_` models if your project requires them) should select from `source`s.
+* All other models should only select from other models.
+* For more information on how we structure our projects see [this document](link to be created)
+
+## Testing
+
+- Every subdirectory should contain a `schema.yml` file, in which each model in the subdirectory is tested.
+- At a minimum, unique and foreign key constraints should be tested (if applicable).
 
-## Field Naming Conventions
+## Naming and field conventions
 
-- TBD
+* Schema, table and column names should be in `snake_case`.
+* Use names based on the _business_ terminology, rather than the source terminology.
+* Table names should be plurals, e.g. `accounts`.
+* Each `stg_` and `fct_` model should have a primary key.

```

Comment:
I believe that _each_ model should have a primary key. When should we _not_ have one?

Reply:
Hot take, but I think that constructing a surrogate key is always best practice.

Let's think about `metrics` where we want one record per account per day.
We would normally construct a surrogate key, like so:
```
{{ dbt_utils.surrogate_key('date_day::date', 'account') }} as id
```
Let's also say that we have a table of targets, also at the grain of one record per day, surrogate key the same way.

For me, the join that looks like this:
```
from metrics
left join targets using (account, date_day)
```
Is actually easier to understand what is _actually_ going on, compared to:
```
from metrics
left join targets using (id)
```
Also the second join won't work if you happen to cast the `date_day` as a `date` in one surrogate key, but a `timestamp` in another.

Yes, any models that are queried in BI tools _need_ a primary key (otherwise Looker won't let you do some things).
But I think constructing surrogate keys in intermediate models (as we might do when we create a model for every metric that a company wants to track) is kind of overkill.

Reply:
strong disagree! I get where you're coming from, but I really do believe that "every model gets a unique ID" is the kind of simple rule that makes it very easy to understand and follow. no judgment required--just do it.

I'm not actually advocating for this to be added as a standard (_yet_) but I definitely want to remove the "this is not good modeling practice" line in your PR above.

Final point: the two-column join is always less performant than the one-column join. Optimizing for readability is a good default unless there are meaningfully negative silicon implications for that decision, which there are here.